File: src/main/java/net/maple/handlers/misc/DropPickUpRequestHandler.java
package net.maple.handlers.misc;

import client.Character;
import client.Client;
import field.obj.FieldObjectType;
import field.obj.drop.AbstractFieldDrop;
import net.maple.handlers.PacketHandler;
import util.packet.PacketReader;

public class DropPickUpRequestHandler extends PacketHandler {

    @Override
    public void handlePacket(PacketReader reader, Client c) {
        Character chr = c.getCharacter();
        reader.readByte();
        reader.readInteger(); // timestamp?
        reader.readShort(); // start of position?
        reader.readShort();
        int id = reader.readInteger();
        reader.readInteger();

        AbstractFieldDrop drop = (AbstractFieldDrop) chr.getField().getObject(FieldObjectType.DROP, id);
        if (drop != null) {
            drop.pickUp(chr);
        } else {
            System.err.println("Picked up drop is null");
        }
    }
}

File: src/main/java/net/maple/handlers/misc/DropPickUpRequestHandler.kt
package net.maple.handlers.misc

import client.Client
import field.obj.drop.AbstractFieldDrop
import net.maple.handlers.PacketHandler
import util.packet.PacketReader

class DropPickUpRequestHandler : PacketHandler {

    override fun handlePacket(reader: PacketReader, c: Client) {
        val chr = c.character
        reader.readByte()
        reader.readInteger() // timestamp?
        reader.readShort() // start of position?
        reader.readShort()
        val id = reader.readInteger()
        reader.readInteger()

        val drop = chr.field.getObject<AbstractFieldDrop>(id) ?: return
        drop.pickUp(chr)
    }
}